### PR TITLE
Move "good" badge helpers from lib/ to services/

### PR DESCRIPTION
**Pull request by paulmelnikow**
This moves a few helpers from `lib/` to `services/`:

build-status.js
build-status.spec.js
color-formatters.js
color-formatters.spec.js
contributor-count.js
licenses.js
licenses.spec.js
php-version.js
php-version.spec.js
text-formatters.js
text-formatters.spec.js
version.js
version.spec.js

And one from `lib/` to `core/`:

unhandled-rejection.spec.js

The diff is long, but the changes are straightforward.

Ref #2832

### CODE REVIEW

**Comment by paulmelnikow** [comment]
This will keep causing merge conflicts as service are changed. It would be great to get this merged. Have added a list of affected files to the top post.

**Comment by calebcartwright** [comment]
I'm can look at this one tonight

**Comment by calebcartwright** [comment]
Was debating whether to ask this here or back over on #2832, but how are we qualifying "good" and "bad" for helper functions? 

(not implying any of the functions in this PR are "bad", just not sure what puts a helper into a category)

File: services/pub/pub.service.js [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 const Joi = require('joi')
 const { BaseJsonService } = require('..')
-const { renderVersionBadge } = require('../../lib/version')
-const { latest: latestVersion } = require('../../lib/version')
+const { renderVersionBadge } = require('../version')
+const { latest: latestVersion } = require('../version')
```
Comment (calebcartwright) [inline services/pub/pub.service.js]: I know this was already in place, but do we really need two separate require statements for `'../version'`?

**Comment by paulmelnikow** [comment]
> Is "bad" just helper functions that are for legacy code/that will be unneeded once the service refactor is complete?

Yes, and `regular-update` which needs a total rewrite for async, maybe using [node-cache](https://github.com/mpneuried/nodecache).

File: services/pub/pub.service.js (hunk 1 above)
Comment (paulmelnikow) [reply services/pub/pub.service.js]: Heh, updated!

File: services/chrome-web-store/chrome-web-store.service.js [hunk 2]
```diff
@@ -5,16 +5,13 @@ const {
...
   downloadCount: downloadCountColor,
   floorCount: floorCountColor,
   version: versionColor,
-} = require('../../lib/color-formatters')
-const {
-  addv: versionText,
-  currencyFromCode,
-} = require('../../lib/text-formatters')
+} = require('../color-formatters')
+const { addv: versionText, currencyFromCode } = require('../text-formatters')
```
Comment (calebcartwright) [inline services/chrome-web-store/chrome-web-store.service.js]: OT comment, but could combine imports for `'../text-formatters'` as well

File: services/pub/pub.service.js (hunk 1 above)
Comment (calebcartwright) [reply services/pub/pub.service.js]: Lol I see you've already changed this

File: services/chrome-web-store/chrome-web-store.service.js (hunk 2 above)
Comment (calebcartwright) [reply services/chrome-web-store/chrome-web-store.service.js]: Given #3114 I don't think this matters here

File: doc/TUTORIAL.md [hunk 3]
```diff
@@ -96,32 +93,35 @@ Each service has a directory for its files:
 All service badge classes inherit from [BaseService] or another class which extends it.
 Other classes implement useful behavior on top of [BaseService].
 
-* [BaseJsonService](https://github.com/badges/shields/blob/master/services/base-json.js)
+- [BaseJsonService](https://github.com/badges/shields/blob/master/services/base-json.js)
```
Comment (calebcartwright) [inline doc/TUTORIAL.md]: Ah just realized these links are bad now, should be `https://github.com/badges/shields/blob/master/core/base-service/base-json.js`

File: doc/TUTORIAL.md [hunk 4]
```diff
@@ -96,32 +93,35 @@ Each service has a directory for its files:
...
   implements methods for performing requests to a JSON API and schema validation.
-* [BaseXmlService](https://github.com/badges/shields/blob/master/services/base-xml.js)
+- [BaseXmlService](https://github.com/badges/shields/blob/master/services/base-xml.js)
```
Comment (calebcartwright) [inline doc/TUTORIAL.md]: should be `https://github.com/badges/shields/blob/master/core/base-service/base-xml.js`

File: doc/TUTORIAL.md [hunk 5]
```diff
@@ -96,32 +93,35 @@ Each service has a directory for its files:
...
   implements methods for performing requests to an XML API and schema validation.
-* If you are contributing to a *service family*, you may define a common super
+- If you are contributing to a _service family_, you may define a common super
   class for the badges or one may already exist.
 
-[BaseService]: https://github.com/badges/shields/blob/master/services/base.js
+[baseservice]: https://github.com/badges/shields/blob/master/services/base.js
```
Comment (calebcartwright) [inline doc/TUTORIAL.md]: should be `https://github.com/badges/shields/blob/master/core/base-service/base.js`

File: doc/TUTORIAL.md [hunk 6]
```diff
@@ -96,32 +93,35 @@ Each service has a directory for its files:
...
 ```js
-'use strict'                                         // (1)
+'use strict' // (1)
 
-const { BaseService } = require('..')                // (2)
+const { BaseService } = require('..') // (2)
 
-module.exports = class Example extends BaseService { // (3)
+module.exports = class Example extends BaseService {
+  // (3)
```
Comment (calebcartwright) [inline doc/TUTORIAL.md]: I'm still not a fan of how prettier moves these comments around (especially num. 3) due to their relation to the bullets below. 

Maybe we could move all the numbered comments to above the line the reference to both be consistent and satisfy prettier?

This doesn't need to be a blocker IMO (given all the changes in this PR) but if we merge these Tutorial changes as-is then I do feel we should update these inline code blocks soon

Comment (paulmelnikow) [reply doc/TUTORIAL.md]: Ahhh my editor must have done that. Thanks for catching it.

Do you think the inline numbers would look okay? They would be a _lot_ easier to maintain, and it'd let us stop prettier-ignoring this file.

Comment (paulmelnikow) [reply doc/TUTORIAL.md]: How about we merge this and one of us can deal with the broken links and formatting as a follow-on?

Comment (calebcartwright) [reply doc/TUTORIAL.md]: > How about we merge this and one of us can deal with the broken links and formatting as a follow-on?

Works for me!

Comment (calebcartwright) [reply doc/TUTORIAL.md]: I opened https://github.com/badges/shields/issues/3116 just to make sure we don't accidentally forget

**Comment by paulmelnikow** [comment]
Thanks for the careful review!

**Pull request merged**